db/data.py

```diff
@@ -359,6 +360,48 @@ def load_low_level(mbid, offset=0):
         return row[0]
 
 
+def load_many_low_level(recordings):
+    """Uses a single query method to collect low-level JSON data for multiple recordings.
+
+    Args:
+        recordings: A list of tuples (mbid, offset).
+
+    Returns:
+
+        {"mbid1": {"offset1": {lowlevel_data},
+                  "offsetn": {lowlevel_data}}
+         ...
+         ...
+
+        "mbidn": {"offset1": {lowlevel_data}}
+        }
+    
+    """
+    
+    with db.engine.connect() as connection:
+        query = text("""
+            SELECT gid, submission_offset, data
+              FROM (
+            SELECT id, gid,
+        ROW_NUMBER () 
+              OVER (PARTITION BY gid 
+          ORDER BY submitted DESC) - 1 submission_offset
+              FROM lowlevel
+            )   AS partitions
+         LEFT JOIN lowlevel_json 
+             USING (id)
+             WHERE (gid, submission_offset) IN :recordings
+        """)
+        
+        result = connection.execute(query, { 'recordings': tuple(recordings) })
+
+        recordings_info = {}
```

Comment:
you can use `defaultdict(dict)` here, then you dont need to use setdefault below